### PR TITLE
Add preDeleteAction to ory reconciler

### PR DESCRIPTION
**Pull request by raypinto**
See https://github.com/kyma-incubator/reconciler/issues/342

### CODE REVIEW

File: pkg/reconciler/instances/ory/ory.go [hunk 1]
```diff
@@ -18,10 +18,13 @@ func init() {
 	}
 
 	reconciler.
-		WithPreReconcileAction(&preAction{
+		WithPreReconcileAction(&preInstallAction{
```
Comment (Tomasz-Smelcerz-SAP) [inline pkg/reconciler/instances/ory/ory.go:21]: The name change implies "installation" operation. Will the same code also handle upgrades? If so, then the new name isn't really good.

File: pkg/reconciler/instances/ory/action.go [hunk 2]
```diff
@@ -163,6 +191,27 @@ func (a *postAction) patchSecret(ctx context.Context, client kubernetes.Interfac
...
+		}
+		return false, errors.Wrap(err, "Could not get DB secret")
+	}
+	return true, nil
+}
+
+func (a *preDeleteAction) deleteSecret(ctx context.Context, client kubernetes.Interface, name types.NamespacedName, logger *zap.SugaredLogger) error {
+	err := client.CoreV1().Secrets(name.Namespace).Delete(ctx, name.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to delete the secret")
```
Comment (Tomasz-Smelcerz-SAP) [inline pkg/reconciler/instances/ory/action.go:208]: I don't know the API, but perhaps it's worth to check if `if kerrors.IsNotFound(err)` and if so, return true here.
It is one of the common network programming errors: You assume that between the check if the secret exists (time == t[0]) until the moment you actually request a delete (time == t[1] == t[0]+delta), nothing changes.
What if your delta is 10 seconds? Easily possible in network scenarios. Ordinary user can delete the secret by hand within that time.

File: pkg/reconciler/instances/ory/ory.go (hunk 1 above)
Comment (raypinto) [reply pkg/reconciler/instances/ory/ory.go:21]: That's a good point. It is used for upgrades as well. I will rename it to preReconcileAction and postReconcileAction respectively.

File: pkg/reconciler/instances/ory/action.go (hunk 2 above)
Comment (raypinto) [reply pkg/reconciler/instances/ory/action.go:211]: It's an edge case. But a valid one. Will do.

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Pull request merged**